### ISSUE TITLE
Method not allowed (GET) on /application/o/token

### ISSUE THREAD

**Issue by ToshY** [goauthentik/authentik#3548]
**Describe your question**

Trying to setup Authentik for [Outline](https://github.com/outline/outline) with [OIDC](https://app.getoutline.com/share/770a97da-13e5-401e-9f8a-37949c19f97e/doc/oidc-8CPBm6uC0I), but it's returning errors in logging on both sides. I cannot login at Outline, which returns error that the authentication failed (see screenshot), but at the same time it shows in Authentik logging `Method Not Allowed (GET): /application/o/token/`. 

As I'm not sure myself if Authentik or Outline is doing something incorrectly, I labeled this as a question. If it ends up being Outline, I'll go file a bug report in their repo. 

**Screenshots**

_Outline_

![image](https://user-images.githubusercontent.com/31921460/188739783-a51f0eb0-ad7e-4892-883f-ce01da80143c.png)

**Logs**
Authentik:

```
authentik-authentik-server-1  | 2022-09-06T21:29:00.829617887Z {"auth_via": "unauthenticated", "event": "updated scope", "host": "authentik.example.com", "level": "debug", "logger": "authentik.providers.oauth2.views.userinfo", "pid": 21, "request_id": "5efb5164b4704c9f9040c6d37752ec5d", "scope": "<ScopeMapping: Scope Mapping authentik default OAuth Mapping: OpenID 'email' (email)>", "timestamp": "2022-09-06T21:29:00.829147"}
authentik-authentik-server-1  | 2022-09-06T21:29:00.833234355Z {"auth_via": "unauthenticated", "event": "updated scope", "host": "authentik.example.com", "level": "debug", "logger": "authentik.providers.oauth2.views.userinfo", "pid": 21, "request_id": "5efb5164b4704c9f9040c6d37752ec5d", "scope": "<ScopeMapping: Scope Mapping authentik default OAuth Mapping: OpenID 'openid' (openid)>", "timestamp": "2022-09-06T21:29:00.832696"}
authentik-authentik-server-1  | 2022-09-06T21:29:00.837452247Z {"auth_via": "unauthenticated", "event": "updated scope", "host": "authentik.example.com", "level": "debug", "logger": "authentik.providers.oauth2.views.userinfo", "pid": 21, "request_id": "5efb5164b4704c9f9040c6d37752ec5d", "scope": "<ScopeMapping: Scope Mapping authentik default OAuth Mapping: OpenID 'profile' (profile)>", "timestamp": "2022-09-06T21:29:00.837142"}
...
authentik-authentik-server-1  | 2022-09-06T21:29:01.833552738Z {"auth_via": "unauthenticated", "event": "/application/o/token/", "host": "authentik.example.com", "level": "info", "logger": "authentik.asgi", "method": "POST", "pid": 21, "remote": "12.34.56.789", "request_id": "5efb5164b4704c9f9040c6d37752ec5d", "runtime": 1887, "scheme": "https", "status": 200, "timestamp": "2022-09-06T21:29:01.833069", "user": "", "user_agent": "Node-oauth"}
authentik-authentik-server-1  | 2022-09-06T21:29:01.980422576Z {"event": "Method Not Allowed (GET): /application/o/token/", "level": "warning", "logger": "django.request", "timestamp": 1662499741.980217}
authentik-authentik-server-1  | 2022-09-06T21:29:01.982397558Z {"auth_via": "unauthenticated", "event": "/application/o/token/", "host": "authentik.example.com", "level": "info", "logger": "authentik.asgi", "method": "GET", "pid": 21, "remote": "12.34.56.789", "request_id": "d5ef091b892d4c7d8903023a17b91ac0", "runtime": 20, "scheme": "https", "status": 405, "timestamp": "2022-09-06T21:29:01.982105", "user": "", "user_agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"}
authentik-authentik-server-1  | 2022-09-06T21:29:01.992364923Z {"event": "Method Not Allowed: /application/o/token/", "level": "warning", "logger": "django.request", "timestamp": 1662499741.9920964}
```

Outline:
```
outline-outline-1          | 2022-09-06T21:29:02.021711132Z {"error":"invalid json response body at https://authentik.example.com/application/o/token/ reason: Unexpected end of JSON input","stack":"FetchError: invalid json response body at https://authentik.example.com/application/o/token/ reason: Unexpected end of JSON input\n    at /opt/outline/node_modules/node-fetch/lib/index.js:273:32\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async OAuth2Strategy._passportOauth.Strategy.userProfile (/opt/outline/build/server/routes/auth/providers/oidc.js:45:22)","level":"error","message":"Error during authentication"}
```

(cant get more logging out of it)

**Version and Deployment (please complete the following information):**
 - authentik version: 2022.8 (2022.8.2)
 - outline version: 0.66.0
 - Deployment: docker compose v2.10.2

**Additional context**
N/A

**Comment by BeryJu** [comment]
authentik's OAuth2 /token/ endpoint only supports POST requests as https://www.rfc-editor.org/rfc/rfc6749#section-3.2 states 
```
   The client MUST use the HTTP "POST" method when making access token
   requests.
```

Could you post your current config?

**Comment by ToshY** [comment]
Hey @BeryJu ,

Could you clarify which config you are referring to? I can post everything from application, provider, event logs, environment variables for docker compose setup.

**Comment by BeryJu** [comment]
Sorry I meant the config you have for outline

**Comment by ToshY** [comment]
Hey @BeryJu 

The moment I copied the environment variables in here to show them, I noticed a stupid (copy-paste) mistake.

```
OUTLINE_OIDC_TOKEN_URI="https://authentik.example.com/application/o/token/"
OUTLINE_OIDC_USERINFO_URI="https://authentik.example.com/application/o/token/"
```

🤦‍♂️  It all makes sense now that it tries to get the user info but throwing the error in the logging as its using GET on the wrong endpoint. After changing it to `/userinfo` it worked.

Sorry for this silly mistake. I appreciate you being my rubber 🦆 here 😄
